test/e2e/backward_compat_test.go

```diff
@@ -113,10 +106,27 @@ var _ = Describe("When upgrading AAD Pod Identity", func() {
 			IdentityClientID: identityClientID,
 		})
 
+		By("Deleting the ConfigMap used to store upgrade information")
+		err = exec.KubectlDelete(kubeconfigPath, framework.NamespaceKubeSystem, []string{
+			"--ignore-not-found",
+			"cm",
+			"aad-pod-identity-config",
+			fmt.Sprintf("--namespace=%s", framework.NamespaceKubeSystem),
+		})
+		Expect(err).To(BeNil())
+
 		helm.Upgrade(helm.UpgradeInput{
 			Config: config,
 		})
 
+		azureassignedidentity.Wait(azureassignedidentity.WaitInput{
```

Comment:
The latest MIC will update the `AzureAssignedIdentity` to `Assigned` state after the type upgrade is complete. So we should be able to rely on that before the actual validation making it more deterministic.